src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -261,6 +261,50 @@ Alias Visitor::visit(Parser::AliasWithoutBracketsContext* ctx) {
   return {visitExpressionPimpl(ctx->expression()), visit(ctx->var())};
 }
 
+// ____________________________________________________________________________________
+parsedQuery::BasicGraphPattern Visitor::toGraphPattern(
+    const ad_utility::sparql_types::Triples& triples) {
+  parsedQuery::BasicGraphPattern pattern{};
+  pattern._triples.reserve(triples.size());
+  for (const auto& triple : triples) {
+    auto toTripleComponent = []<typename T>(const T& item) {
+      namespace tc = ad_utility::triple_component;
+      if constexpr (ad_utility::isSimilar<T, Variable>) {
+        return TripleComponent{item};
+      } else if constexpr (ad_utility::isSimilar<T, Literal>) {
+        return TripleComponent{
+            tc::Literal::fromStringRepresentation(item.toSparql())};
+      } else if constexpr (ad_utility::isSimilar<T, Iri>) {
+        return TripleComponent{
+            tc::Iri::fromStringRepresentation(item.toSparql())};
```

Comment:
I am not sure if this is correct, as the `fromStringRepresentation` is already unescaped in its contents etc, which to `toSParql()` is probably not.
Can you please check this out (with escape characters in strign literals etc.) 
Independent of that a worthwile next step here would be to get rid of the `Literal` and `Iri` type, and consistently only use `TripleComponent::Literal` and `TripleComponent::Iri`. Then this function becomes trivial.

Reply:
As discussed, feed them to the `TurtleParser::parseTripleObject` function.